docs/csharp/reference-semantics-with-value-types.md

```diff
@@ -88,17 +88,40 @@ temporary variable created as part of the method call.
 There are several ways in which the compiler ensures that the read-only
 nature of an `in` argument is enforced.  First of all, the called method
 can't directly assign to an `in` parameter. It can't directly assign
-to any field of an `in` parameter. In addition, you cannot pass
-an `in` parameter to any method demanding the `ref` or `out` modifier.
-The compiler enforces that the `in` argument is a readonly variable. You
-can call any instance method that uses pass-by-value semantics. In
+to any field of an `in` parameter when that value is a `struct` type. In addition, you cannot pass
+an `in` parameter to any method using the `ref` or `out` modifier.
+These rules apply to any field of an `in` parameter provided the
+field is a `struct` type and the parameter is also a `struct` type. In fact, these rules
+apply for multiple layers of member access provided the types at all levels
+of member access are `structs`.
+The compiler enforces that `struct` types passed as  `in` arguments and their
+`struct` members are readonly variables.
+
+The use of `in` parameters is to avoid the potential performance costs
+of copies. It does not change the semantics of any method call. Therefore,
+you do not need to specify the `in` modifier at the call site. However,
+omitting the `in` modifier at the call site informs the compiler that it is
+allowed to make a copy of the argument for the following reasons:
+
+- There is an implicit conversion but not an identity conversion from the argument type to the parameter type.
+- The argument is an expression but does not have a known storage variable.
+- An overload exists that differs by the presence or absence of `in`. In that case, the by value overload is a better match.
+
+These rules are useful as you update existing code to use readonly
+reference arguments. Inside the called method, you can call any instance
+method that uses by value parameters. In
 those instances, a copy of the `in` parameter is created. Because the compiler
 can create a temporary variable for any `in` parameter, you can also specify default
-values for any `in` parameter. The follow code uses that to specify the origin
+values for any `in` parameter. The following code uses that to specify the origin
 (point 0,0) as the default value for the second point:
 
 [!code-csharp[InArgumentDefault](../../samples/csharp/reference-semantics/Program.cs#InArgumentDefault "Specifying defaults for an in parameter")]
 
+To force the compiler to pass readonly arguments by reference, specify the `in` modifer
+on the arguments at the callsite, as shown in the following code:
+
+[!code-csharp[UseInArgument](../../samples/csharp/reference-semantics/Program.cs#ExplicitInArgument "Specifying an In argument")]
+
 The `in` parameter designation can also be used with reference types or built in
```

Comment:
built in --> built-in 
what are built-in numeric values? constants? numeric literals?

Reply:
I deleted "built in". (It was taken from the draft spec) It refers to the numeric types: byte, short, int, long, double and so on)